src/python/ensembl/io/genomio/gff3/features.py

```diff
@@ -0,0 +1,50 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""GFF3 features."""
+
+from __future__ import annotations
+
+__all__ = [
+    "GFFSeqFeature",
+]
+
+from typing import List
+
+from Bio.SeqFeature import SeqFeature, Location
+
+
+class GFFSeqFeature(SeqFeature):
+    """Extends `Bio.SeqFeature.SeqFeature` with sub_features, to be used for typing."""
+
+    def __init__(
+        self,
+        location: Location | None = None,
+        type: str = "",  # pylint: disable=W0622
+        id: str = "<unknown id>",  # pylint: disable=W0622
+        qualifiers: dict | None = None,
+        sub_features: List[GFFSeqFeature] | None = None,
```

Comment:
If you use `dict`, then you can use `list` (and avoid the `from typing import List` above) [Python 3.10], or we revert `dict` to `Dict` [Python 3.9]. I can see in the future we will have a mix, but since this is new code, I prefer to adhere to one or the other.